### PR TITLE
Update node spec in Jenkinsfile and Docker config mount

### PR DESCRIPTION
**Pull request by aiordache**
Fix test setup

### CODE REVIEW

**Comment by StefanScherer** [comment]
I've verified that the `config.json` is valid now in the outer and the inner container:

```
ubuntu@ip-10-100-50-213:~$ docker ps
CONTAINER ID   IMAGE                                                                  COMMAND                  CREATED         STATUS         PORTS           NAMES
1366ac199d9f   dockerpinata/compose:debian-6ea24001fad37c8327bca6048e56aabcb27ad543   "tox -e py37 -- --ve…"   6 minutes ago   Up 6 minutes                   goofy_euler
dc7d4533187d   docker:19.03.13-dind                                                   "dockerd-entrypoint.…"   6 minutes ago   Up 6 minutes   2375-2376/tcp   compose-dind-19.03.13-18
64d69e0868f3   dockerpinata/compose:debian-6ea24001fad37c8327bca6048e56aabcb27ad543   "script/test/ci --ve…"   7 minutes ago   Up 7 minutes                   blissful_chatterjee
ubuntu@ip-10-100-50-213:~$ docker exec dc7d cat /root/.docker/config.json
{
	"auths": {
		"https://index.docker.io/v1/": {
			"auth": "redacted"
		}
	}
}ubuntu@ip-10-100-50-213:~$ docker exec 64d6 cat /root/.docker/config.json
{
	"auths": {
		"https://index.docker.io/v1/": {
			"auth": "redacted"
		}
	}
}ubuntu@ip-10-100-50-213:~$ docker exec 64d6 ls -l /root/.docker/config.json
-rw------- 1 1000 1000 139 Apr 15 12:19 /root/.docker/config.json
ubuntu@ip-10-100-50-213:~$ docker exec dc7d ls -l /root/.docker/config.json
-rw-r--r--    1 1000     1000           139 Apr 15 12:19 /root/.docker/config.json
ubuntu@ip-10-100-50-213:~$ 
```

**Comment by StefanScherer** [comment]
A mount with `-v /root/.docker/config.json:/root/.docker/config.json \` didn't work, the inner `/root/.docker/config.json/` was then a folder, maybe because of the restricted file permissions of the Jenkins credential.

**Review by StefanScherer — APPROVED** [review]
LGTM  🔍 🔬

File: Jenkinsfile [hunk 1]
```diff
@@ -87,7 +87,7 @@ def buildImage(baseImage) {
 def runTests(dockerVersion, pythonVersion, baseImage) {
     return {
         stage("python=${pythonVersion} docker=${dockerVersion} ${baseImage}") {
-            node("ubuntu && amd64 && !zfs") {
+            node("ubuntu-2004 && amd64 && !zfs && cgroup1") {
```
Comment (ulyssessouza) [inline Jenkinsfile:90]: Doesn't it imply considering `cgroups2` in the possible nodes?

**Review by ulyssessouza — APPROVED** [review]
LGTM

**Pull request merged**